app/recurring_job.go

```diff
@@ -403,7 +403,8 @@ func (job *Job) doSnapshotCleanup(backupDone bool) (err error) {
 		return err
 	}
 
-	if len(deleteSnapshotNames) > 0 {
+	shouldPurgeSnapshots := len(deleteSnapshotNames) > 0 || job.task == longhorn.RecurringJobTypeSnapshotCleanup
```

Comment:
What if `deleteSnapshotNames` (for task snapshot) is 0 but there are system snapshots? Should we help purge them?

Reply:
Absolutely, update!